In [65]:
import pandas as pd
import numpy as np 

In [66]:
df=pd.read_csv("../data\CineStream_Catalog.csv")

In [67]:
df.head()

,ContentID,Title,Type,Genre,Language,Country,Director,CastCount,ReleaseYear,AddedDate,...,IMDbScore,ViewsMillions,WatchHoursMillions,AvgCompletionPct,AwardsWon,ProductionCostCr,RevenueCr,SubscribersGainedThousands,TrendingStatus,CriticRating
0,CS-0001,Voyage Whispering II,Movie,Kids,Hindi,India,Akira Brown,3,2024,2020-04-28,...,8.0,25.28,40.81,65,4,56.1,59.5,5873,Classic,7.4
1,CS-0002,Endless River: Reborn,Series,Kids,Korean,South Korea,Riya Patel,6,2017,2020-07-30,...,6.8,24.13,244.20,69,0,43.6,33.6,-1000,New,6.2
2,CS-0003,Distant Letter of the North,Series,Romance,Telugu,India,Liam Iyer,4,2016,2020-08-29,...,7.7,26.77,107.08,80,2,63.6,56.6,1016,Popular,6.9
3,CS-0004,Light Whispering,Series,Animation,English,Australia,Min-jun Khan,5,2020,2020-02-16,...,5.6,8.67,111.27,55,0,44.3,8.6,-1000,Trending,6.6
4,CS-0005,Game Quiet III,Stand-up,Comedy,Hindi,India,Yuki Yilmaz,13,2017,2021-04-20,...,5.5,7.01,5.29,52,0,1.7,10.6,1976,Classic,4.8


In [68]:
df.shape

(524, 23)

In [69]:
df.columns

Index(['ContentID', 'Title', 'Type', 'Genre', 'Language', 'Country',
       'Director', 'CastCount', 'ReleaseYear', 'AddedDate', 'RuntimeMinutes',
       'EpisodeCount', 'AgeRating', 'IMDbScore', 'ViewsMillions',
       'WatchHoursMillions', 'AvgCompletionPct', 'AwardsWon',
       'ProductionCostCr', 'RevenueCr', 'SubscribersGainedThousands',
       'TrendingStatus', 'CriticRating'],
      dtype='object')

In [70]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 524 entries, 0 to 523
Data columns (total 23 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   ContentID                   524 non-null    object 
 1   Title                       524 non-null    object 
 2   Type                        524 non-null    object 
 3   Genre                       524 non-null    object 
 4   Language                    524 non-null    object 
 5   Country                     524 non-null    object 
 6   Director                    520 non-null    object 
 7   CastCount                   524 non-null    int64  
 8   ReleaseYear                 524 non-null    int64  
 9   AddedDate                   521 non-null    object 
 10  RuntimeMinutes              524 non-null    int64  
 11  EpisodeCount                524 non-null    int64  
 12  AgeRating                   524 non-null    object 
 13  IMDbScore                   524 non

In [71]:
df.isnull().sum()

ContentID                     0
Title                         0
Type                          0
Genre                         0
Language                      0
Country                       0
Director                      4
CastCount                     0
ReleaseYear                   0
AddedDate                     3
RuntimeMinutes                0
EpisodeCount                  0
AgeRating                     0
IMDbScore                     0
ViewsMillions                 0
WatchHoursMillions            0
AvgCompletionPct              0
AwardsWon                     0
ProductionCostCr              0
RevenueCr                     0
SubscribersGainedThousands    0
TrendingStatus                0
CriticRating                  0
dtype: int64

In [72]:
df.duplicated().sum()

np.int64(4)

In [73]:
df.value_counts()

ContentID  Title                         Type         Genre      Language  Country      Director       CastCount  ReleaseYear  AddedDate   RuntimeMinutes  EpisodeCount  AgeRating  IMDbScore  ViewsMillions  WatchHoursMillions  AvgCompletionPct  AwardsWon  ProductionCostCr  RevenueCr  SubscribersGainedThousands  TrendingStatus  CriticRating
CS-0211    Silent Mountain II            Movie        Action     Turkish   Turkey       Diego Jones    10         2017         2022-10-18  159             1             U/A 13+    6.3        20.00          34.45               65                2          27.0              45.6        4630                       New             4.8             2
CS-0207    Bridge Bright                 Documentary  Romance    Hindi     India        Aylin Yilmaz   15         2022         2023-08-21  66              1             U          7.5        12.18          9.51                71                2          10.7              20.2        2185                       N

In [74]:
categorical_cols = [
    "Type",
    "Genre",
    "Language",
    "Country",
    "AgeRating",
    "TrendingStatus"
]

for col in categorical_cols:
    print(f"\n{col}")
    print(df[col].unique())


Type
['Movie' 'Series' 'Stand-up' 'movie' 'Documentary' 'series' 'documentary']

Genre
['Kids' 'Romance' 'Animation' 'Comedy' 'Action' 'Sci-Fi' 'Thriller'
 'Reality' 'Drama' 'Action ' 'Horror' 'Romance ' 'Drama ' 'Comedy '
 'Reality ' 'Horror ' 'Thriller ' 'Kids ' 'Sci-Fi ']

Language
['Hindi' 'Korean' 'Telugu' 'English' 'Tamil' 'Turkish' 'Bengali'
 'Malayalam' 'Hindi ' 'Kannada' 'Spanish' 'Japanese' 'Telugu ' 'Tamil '
 'English ' 'Turkish ' 'Kannada ']

Country
['India' 'South Korea' 'Australia' 'Turkey' 'UK' 'USA' 'Spain' 'Canada'
 'Japan']

AgeRating
['U/A 13+' 'U/A 7+' 'U/A 16+' 'U' 'A']

TrendingStatus
['Classic' 'New' 'Popular' 'Trending']


## Dataset Overview

- Total Rows: 524
- Total Columns: 23

## Missing Values

- Director column has 4 missing values.
- AddedDate column has 3 missing values.

## Duplicate Rows

- 4 duplicate rows found in the dataset.

## Issues Found

### Type Column
- Case mismatch found.
- Examples: Movie, movie, Series, series, Documentary, documentary.

### Genre Column
- Trailing spaces found.
- Examples: Drama and Drama , Action and Action .

### Language Column
- Trailing spaces found.
- Examples: Hindi and Hindi , Tamil and Tamil .

### IMDbScore
- Need to check for values outside the valid range (1 to 10).

### CriticRating
- Need to check for values outside the valid range (1 to 10).

### RuntimeMinutes
- Need to check for unusual or very large runtime values.

### SubscribersGainedThousands
- Need to check for negative values.



In [75]:
df = df.drop_duplicates().copy()

In [76]:
text_cols = [
    "Type",
    "Genre",
    "Language",
    "Country",
    "Director",
    "AgeRating",
    "TrendingStatus"
]

for col in text_cols:
    df[col] = df[col].astype(str).str.strip()

In [77]:
df["Type"] = df["Type"].str.title()

In [78]:
df["AddedDate"] = pd.to_datetime(
    df["AddedDate"],
    errors="coerce"
)

In [79]:
df["Profit_Cr"] = (
    df["RevenueCr"]
    - df["ProductionCostCr"]
)


In [80]:
df["ROI_Pct"] = (
    df["Profit_Cr"]
    / df["ProductionCostCr"]
) * 100

In [81]:
df["Performance_Band"] = np.where(
    df["Profit_Cr"] > 20,
    "Hit",
    np.where(
        df["Profit_Cr"] >= 0,
        "Break-even",
        "Flop"
    )
)

In [82]:
df.head()

,ContentID,Title,Type,Genre,Language,Country,Director,CastCount,ReleaseYear,AddedDate,...,AvgCompletionPct,AwardsWon,ProductionCostCr,RevenueCr,SubscribersGainedThousands,TrendingStatus,CriticRating,Profit_Cr,ROI_Pct,Performance_Band
0,CS-0001,Voyage Whispering II,Movie,Kids,Hindi,India,Akira Brown,3,2024,2020-04-28,...,65,4,56.1,59.5,5873,Classic,7.4,3.4,6.060606,Break-even
1,CS-0002,Endless River: Reborn,Series,Kids,Korean,South Korea,Riya Patel,6,2017,2020-07-30,...,69,0,43.6,33.6,-1000,New,6.2,-10.0,-22.935780,Flop
2,CS-0003,Distant Letter of the North,Series,Romance,Telugu,India,Liam Iyer,4,2016,2020-08-29,...,80,2,63.6,56.6,1016,Popular,6.9,-7.0,-11.006289,Flop
3,CS-0004,Light Whispering,Series,Animation,English,Australia,Min-jun Khan,5,2020,2020-02-16,...,55,0,44.3,8.6,-1000,Trending,6.6,-35.7,-80.586907,Flop
4,CS-0005,Game Quiet III,Stand-Up,Comedy,Hindi,India,Yuki Yilmaz,13,2017,2021-04-20,...,52,0,1.7,10.6,1976,Classic,4.8,8.9,523.529412,Break-even


In [83]:
df.loc[(df['IMDbScore'] < 1.0) | (df['IMDbScore'] > 10.0), 'IMDbScore'] = np.nan

In [84]:
df.loc[(df['CriticRating'] < 1.0) | (df['CriticRating'] > 10.0), 'CriticRating'] = np.nan

In [85]:
df.loc[df['SubscribersGainedThousands'] < 0, 'SubscribersGainedThousands'] = 0

In [86]:
df.loc[df['RuntimeMinutes'] > 1000, 'RuntimeMinutes'] = np.nan

In [87]:
df.to_csv("../outputs/cleaned_cinestream.csv",index=False)